src/ai-did-you-mean-this/azext_ai_did_you_mean_this/custom.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from http import HTTPStatus
+
+import requests
+from requests import RequestException
+
+import azure.cli.core.telemetry as telemetry
+
+from knack.log import get_logger
+from knack.util import CLIError  # pylint: disable=unused-import
+
+from azext_ai_did_you_mean_this.failure_recovery_recommendation import FailureRecoveryRecommendation
+from azext_ai_did_you_mean_this._style import style_message
+from azext_ai_did_you_mean_this._const import (
+    RECOMMENDATION_HEADER_FMT_STR,
+    UNABLE_TO_HELP_FMT_STR,
+    TELEMETRY_MUST_BE_ENABLED_STR,
+    TELEMETRY_MISSING_SUBSCRIPTION_ID_STR,
+    TELEMETRY_MISSING_CORRELATION_ID_STR,
+    UNABLE_TO_CALL_SERVICE_STR
+)
+from azext_ai_did_you_mean_this._cmd_table import CommandTable
+
+logger = get_logger(__name__)
+
+
+# Commands
+# note: at least one command is required in order for the CLI to load the extension.
+def show_extension_version():
+    print(f'Current version: 0.1.0')
+
+
+def _log_debug(msg, *args, **kwargs):
+    # TODO: see if there's a way to change the log formatter locally without printing to stdout
+    msg = f'[Thoth]: {msg}'
+    logger.debug(msg, *args, **kwargs)
+
+
+def get_parameter_table(cmd_table, command, recurse=True):
+    az_cli_command = cmd_table.get(command, None)
+    parameter_table = az_cli_command.arguments if az_cli_command else None
+
+    # if the specified command was not found and recursive search is enabled...
+    if not az_cli_command and recurse:
+        # if there are at least two tokens separated by whitespace, remove the last token
+        last_delim_idx = command.rfind(' ')
+        if last_delim_idx != -1:
+            _log_debug('Removing unknown token "%s" from command.', command[last_delim_idx + 1:])
+            # try to find the truncated command.
+            parameter_table, command = get_parameter_table(cmd_table, command[:last_delim_idx], recurse=False)
+
+    return parameter_table, command
+
+
+def normalize_and_sort_parameters(cmd_table, command, parameters):
+    from knack.deprecation import Deprecated
+    _log_debug('normalize_and_sort_parameters: command: "%s", parameters: "%s"', command, parameters)
+
+    parameter_set = set()
+    parameter_table, command = get_parameter_table(cmd_table, command)
+
+    if parameters:
+        # TODO: Avoid setting rules for global parameters manually.
+        rules = {
+            '-h': '--help',
+            '-o': '--output',
+            '--only-show-errors': None,
+            '--help': None,
+            '--output': None,
+            '--query': None,
+            '--debug': None,
+            '--verbose': None
+        }
+
+        blacklisted = {'--debug', '--verbose'}
```

Comment:
Suggest to use `blocklist` just to avoid offensive language trouble.

Reply:
See https://bugs.chromium.org/p/chromium/issues/detail?id=981129

Reply:
Changed `blacklisted` to `blocklisted`.